scripts/doctest.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# Install doctest's Cabal integration
+cabal install doctest --flag cabal-doctest --ignore-project --overwrite-policy=always
```

Comment:
Could you try moving this part to the project build stage? This would speed things up, since we would no longer need to build `doctest` for every package from scratch, we would only build it once for every ghc version

Reply:
Good idea. I just realized that the `cabal-targets.hs` script also does some not insignificant building, so I'll try to pre-build that too.

Reply:
I checked the log timestamps for `cabal-targets.hs` and it seems to take no time at all, perhaps because we're already running `check-workflow-test-matrix.hs` in the build stage, and that has all of the dependencies.

It seems that `check-workflow-test-matrix.hs` isn't causing any dependencies to be built either, but I assume that's because they get included in our cache.